src/app/backend/resource/pod/poddetail.go

```diff
@@ -112,11 +143,56 @@ func GetPodDetail(client k8sClient.Interface, heapsterClient client.HeapsterClie
 	}
 	configMapList := <-channels.ConfigMapList.List
 
-	podDetail := toPodDetail(pod, metrics, configMapList)
+	podDetail := toPodDetail(pod, metrics, configMapList, creator)
 	return &podDetail, nil
 }
 
-func toPodDetail(pod *api.Pod, metrics []metric.Metric, configMaps *api.ConfigMapList) PodDetail {
+func getPodCreator(client k8sClient.Interface, creatorAnnotation string, nsQuery *common.NamespaceQuery, heapsterClient client.HeapsterClient) (*Creator, error) {
+	var serializedReference api.SerializedReference
+	err := json.Unmarshal([]byte(creatorAnnotation), &serializedReference);
+	if err != nil {
+		return nil, err
+	}
+
+	channels := &common.ResourceChannels{
+		PodList:   common.GetPodListChannel(client, nsQuery, 1),
+		EventList: common.GetEventListChannel(client, nsQuery, 1),
+	}
+	pods := <-channels.PodList.List
+	if err := <-channels.PodList.Error; err != nil {
+		return nil, err
+	}
+
+	events := <-channels.EventList.List
+	if err := <-channels.EventList.Error; err != nil {
+		return nil, err
+	}
+	reference := serializedReference.Reference
+	kind := reference.Kind
+	switch kind {
```

Comment:
I'm trying to find a solution to avoid switch here, but it seems there aren't any. What about splitting code to separate functions?